clang/test/dpct/helper_files_ref/include/device.hpp

```diff
@@ -640,6 +640,22 @@ static inline unsigned int get_device_id(const sycl::device &dev){
   return dev_mgr::instance().get_device_id(dev);
 }
 
+inline void capability_check(const sycl::device &dev,
+                             const std::initializer_list<sycl::aspect> &props) {
+  static const std::unordered_map<sycl::aspect, std::string> ErrMap{
+      {sycl::aspect::fp64, "double"},
+      {sycl::aspect::fp16, "half"},
+  };
+  for (const auto &it : props) {
+    const auto find = ErrMap.find(it);
+    if (find == ErrMap.end())
```

Comment:
how about switch/case or array/idx to get the error message for each aspect, then no need to lookup in map. (anyway try to remove the map lookup).

Reply:
fixed